lib/phase/flatten.js

```diff
@@ -2,14 +2,41 @@
 
 const Hoek = require('hoek');
 const clone = require('clone');
+const TEMPLATE_REGEX = require('screwdriver-data-schema').config.regex.FULL_TEMPLATE_NAME;
+const NAME_MATCH = 1;
+const VERSION_MATCH = 3;
+const LABEL_MATCH = 7;
 
 /**
- * Overlays the stuff specified in a specific job on top of the defaults in shared
+ * Merge oldJob into newJob
+ * "oldJob" takes precedence over "newJob". For ex: individual job settings vs shared settings
+ * @param  {Object}   newJob    Job to be merged into. For ex: shared settings
+ * @param  {Object}   oldJob    Job to merge. For ex: individual job settings
+ */
+function merge(newJob, oldJob) {
```

Comment:
This function does more than just merging, it also initialize fields it does not have, and remove duplicates if merged.  This is the console log from some tests. 

Using this function:
```
{
  "environment": {
    "NODE_ENV": "test"
  },
  "image": "node:{{NODE_VERSION}}",
  "matrix": {
    "NODE_VERSION": [
      4,
      5,
      6
    ]
  },
  "steps": [
    {
      "init": "npm install"
    },
    {
      "test": "npm test"
    }
  ],
  "settings": {},
  "secrets": []
}
```

Using hoek.merge
```
{
  "environment": {
    "NODE_ENV": "test"
  },
  "image": "node:{{NODE_VERSION}}",
  "matrix": {
    "NODE_VERSION": [
      4,
      5,
      6,
      4,
      5,
      6
    ]
  },
  "steps": [
    {
      "init": "npm install"
    },
    {
      "test": "npm test"
    },
    {
      "init": "npm install"
    },
    {
      "test": "npm test"
    }
  ]
}

```